### ISSUE TITLE
Suporte a Mac com processador ARM

### ISSUE THREAD

**Issue by humbertodias** [DanielMoura79/HAMOOPI#4]
Ótima engine! Parabéns pela iniciativa.

Gostaria de contribuir com o projeto e notei que atualmente utiliza allegro 4 que infelizmente nao é compilável com Mac M1/2 com processadores ARM e surgiu uma dúvida sobre próximos passos antes da contribuiçao:

@DanielMoura79 pretende evoluir para qual direção?

1. Adaptar Allegro 4 com [Allegro Legacy](https://hoten.cc/blog/porting-zelda-classic-to-the-web/#allegro-legacy) mantendo uma camada fake com Allegro 5?
2. Portar [Allegro 4 para 5](https://github.com/liballeg/allegro5/wiki/Porting-Allegro-4-to-5) ?
3. Rescrever com [SDL2](https://www.libsdl.org/)?
4. Rescrever com [Raylib](https://www.raylib.com/)?

* Se optar pela reescrita, aqui uma comparaçao entre [SDL vs Raylib](https://gist.github.com/raysan5/17392498d40e2cb281f5d09c0a4bf798#file-raylib_vs_sdl-md)

Obrigado.

**Comment by xmaxmex** [comment]
Boa tarde,

Seria interessante ir para Allegro, pois portar par SDL2, teria que
reescrever.

Já em Allegro 5 teremos que rescrever algumas estruturas.

Que bom que temos mais gente no projeto.


Em seg, 25 de set de 2023 13:28, Humberto Dias ***@***.***>
escreveu:

**Comment by humbertodiaszup** [comment]
Sim. Enfim, consegui adaptar o HAMOOPI para Mac com processador ARM64.
Foi preciso portar de allegro 4 para 5 e compilar estaticamente, seguindo os passos:
1. Compilar e instalar [Allegro-Legacy](https://github.com/NewCreature/Allegro-Legacy)
2. Compilar e instalar  [Allegro 5](https://github.com/liballeg/allegro5/releases/download/5.2.8.0/allegro-5.2.8.0.tar.gz)
3. Compilar HAMOOPI estaticamente
`g++ HAMOOPI.cpp -o HAMOOPI $(allegro-config --cflags --libs --static= -Wl,-rpath,'@executable_path'`

Executável
[HAMOOPI_Allegro5_Mac_ARM64.zip](https://github.com/DanielMoura79/HAMOOPI/files/12743756/HAMOOPI_Allegro5_Mac_ARM64.zip)
Demo
https://github.com/DanielMoura79/HAMOOPI/assets/136126355/4faa7f5b-87c5-41e4-8111-3376294fa4f3

**Comment by humbertodias** [comment]
Sim, optei pela alternativa 1. 
Consegui adaptar o HAMOOPI para Mac com processador ARM64.
Foi preciso portar de allegro 4 para 5 e compilar estaticamente, seguindo os passos:
1. Compilar e instalar [Allegro-Legacy](https://github.com/NewCreature/Allegro-Legacy)
2. Compilar e instalar  [Allegro 5](https://github.com/liballeg/allegro5/releases/download/5.2.8.0/allegro-5.2.8.0.tar.gz)
3. Compilar HAMOOPI estaticamente
`g++ HAMOOPI.cpp -o HAMOOPI $(allegro-config --cflags --libs --static) -Wl,-rpath,'@executable_path'`

Executável
[HAMOOPI_Allegro5_Mac_ARM64.zip](https://github.com/DanielMoura79/HAMOOPI/files/12743756/HAMOOPI_Allegro5_Mac_ARM64.zip)
Demo
https://github.com/DanielMoura79/HAMOOPI/assets/136126355/4faa7f5b-87c5-41e4-8111-3376294fa4f3

**Comment by humbertodias** [comment]
@xmaxmex fechando issue
substitui allegro por sdl aqui https://github.com/DanielMoura79/HAMOOPI/pull/5 e agora funciona em hardware moderno. 
pode testar, baixando em https://github.com/humbertodias/hamoopi/releases
valeu
